doc/InteroperabilityGuide.md

```diff
@@ -0,0 +1,64 @@
+Use *cupla* together with other alpaka based software
+=====================================================
+
+Sometimes it is necessary to use an `alpaka` function for which there is no equivalent in `cupla`, or you want to use a library with `cupla` that was developed with an `alpaka` compatible interface, such as [vikunja](https://github.com/alpaka-group/vikunja/). This requires some `alpaka` types that are handled automatically and internally by `cupla`. The tutorial shows how to access these types.
+
+Accessing Acc Mapping, Acc Type, Queue Type and Stream
+======================================================
+
+The first code shows how `alpaka` generates the required accelerator types. The second code shows how to get the accelerator types in `cupla`.
+
+```C++
+#include <alpaka/alpaka.hpp>
+
+using Dim = alpaka::DimInt<3>;
+using Idx = std::size_t;
+
+using Host = alpaka::AccCpuSerial<Dim, Idx>;
+using Dev = alpaka::AccGpuCudaRt<Dim, Idx>;
+
+using QueueProperty = alpaka::Blocking;
+
+using HostQueue = alpaka::Queue<Host, QueueProperty>;
+using DevQueue = alpaka::Queue<Dev, QueueProperty>;
+
+auto const hostAcc = alpaka::getDevByIdx<Host>(0u);
+HostQueue hostQueue(hostAcc);
+
+auto const devAcc = alpaka::getDevByIdx<Dev>(0u);
+DevQueue devQueue(devAcc);
+
+// ...
+
+alpaka::exec<Dev>(
+	devQueue,
+	workDiv,
+	kernel);
+
+
+```
+
+```C++
+#include <cupla.hpp>
+#include <alpaka/alpaka.hpp>
+
+cupla::AccHost const &hostAcc(cupla::manager::Device<cupla::AccHost>::get().current());
```

Comment:
Is there a need to keep the device object as a reference? All alpaka objects are handles.

Reply:
I'm not sure. This is chicken-egg problem :sweat_smile: I write the documentation because I'm not sure, if I do it right. I test it with the vikunja code and if it is working, I change it.

Reply:
It's working. Therefore I change to a copy.